### PR TITLE
[Upstream] switch to dynamic gmmlib

### PR DESCRIPTION
**Pull request by yangnin**
Signed-off-by: Yang, Ning <ning.yang@intel.com>

### CODE REVIEW

File: media_driver/media_top_cmake.cmake [hunk 1]
```diff
@@ -81,16 +81,29 @@ bs_ufo_link_libraries_noBsymbolic(
...
 )
 
+find_package(PkgConfig)
+
 if (NOT DEFINED INCLUDED_LIBS OR "${INCLUDED_LIBS}" STREQUAL "")
-    # dep libs (gmmlib for now) can be passed through INCLUDED_LIBS, but if not, we need try to setup dep through including dep projects
-    if (NOT TARGET gmm_umd)
-        add_subdirectory("${BS_DIR_GMMLIB}" "${CMAKE_BINARY_DIR}/gmmlib")
+
+    pkg_check_modules(LIBGMMUMMDLL igdgmm)
```
Comment (dvrogozh) [inline media_driver/media_top_cmake.cmake]: LIBGMMUMMDLL name is awful. Sooner or later someone who will modify cmake files will make a mistake typing it. Let's be simpler: GMM will probably be enough...

File: media_driver/media_top_cmake.cmake [hunk 2]
```diff
@@ -81,16 +81,29 @@ bs_ufo_link_libraries_noBsymbolic(
...
     endif()
 
-    if (TARGET gmm_umd)
-        target_link_libraries ( ${LIB_NAME}
-            gmm_umd
-        )
+    if(NOT LIBGMMUMMDLL_FOUND)
+
+        # dep libs (gmmlib for now) can be passed through INCLUDED_LIBS, but if not, we need try to setup dep through including dep projects
+        if (NOT TARGET igfx_gmmumd_dll)
```
Comment (dvrogozh) [inline media_driver/media_top_cmake.cmake]: I did not understand that code before and I don't understand it now. What this if(NOT TARGET) is about? What if(TARGET) below is about? Comment above don't actually help!

Please, describe what's going on if GMM was not found via pkg-config? You eventually proceed with searching gmmlib in BS_DIR_GMMLIB, but which variant you use: static or dynamic? Either way, commands you try to program don't settle in my mind...

File: media_driver/media_top_cmake.cmake (hunk 1 above)
Comment (yangnin) [reply media_driver/media_top_cmake.cmake]: Good suggestion. I'll change it.

File: media_driver/media_top_cmake.cmake (hunk 2 above)
Comment (yangnin) [reply media_driver/media_top_cmake.cmake]: yes, this code part considers no gmm lib install situation. 
use pkg_check module to check if gmmlib is installed in OS ENV.
if not. cmake will build source instand.  
TARGET is chandged from gmm_umd to igfx_gmmumd_dll

Comment (dvrogozh) [reply media_driver/media_top_cmake.cmake]: I still don't understand about TARGET... If pkg-config will fail which variant media-driver will try to build: static or dynamic?

File: media_driver/media_top_cmake.cmake [hunk 3]
```diff
@@ -81,16 +81,28 @@ bs_ufo_link_libraries_noBsymbolic(
...
+    if(NOT GMM_FOUND)
+
+        # dep libs (gmmlib for now) can be passed through INCLUDED_LIBS, but if not, we need try to setup dep through including dep projects
+        if (NOT TARGET igfx_gmmumd_dll)
+            add_subdirectory("${BS_DIR_GMMLIB}" "${CMAKE_BINARY_DIR}/gmmlib")
+        endif()
+
+        if (TARGET igfx_gmmumd_dll)
+            target_link_libraries ( ${LIB_NAME}
+            igfx_gmmumd_dll)
```
Comment (dvrogozh) [inline media_driver/media_top_cmake.cmake]: Should not this be done for the case when pkg-config has detected the gmmlib?

File: media_driver/media_top_cmake.cmake [hunk 4]
```diff
@@ -81,16 +81,28 @@ bs_ufo_link_libraries_noBsymbolic(
...
 if (NOT DEFINED INCLUDED_LIBS OR "${INCLUDED_LIBS}" STREQUAL "")
-    # dep libs (gmmlib for now) can be passed through INCLUDED_LIBS, but if not, we need try to setup dep through including dep projects
-    if (NOT TARGET gmm_umd)
-        add_subdirectory("${BS_DIR_GMMLIB}" "${CMAKE_BINARY_DIR}/gmmlib")
+
+    pkg_check_modules(GMM igdgmm)
+
+    if(GMM_FOUND)
+        include_directories(BEFORE ${GMM_INCLUDE_DIRS})
+        link_directories(${GMM_LIBRARY_DIRS})
```
Comment (dvrogozh) [inline media_driver/media_top_cmake.cmake]: This misses target_link_libraries(), isn't it?

**Pull request closed without merging**